geopandas/geodataframe.py

```diff
@@ -1395,6 +1410,13 @@ def __finalize__(self, other, method=None, **kwargs):
             for name in self._metadata:
                 object.__setattr__(self, name, getattr(other.objs[0], name, None))
 
+            if len(self.columns[self.columns == self._geometry_column_name]) > 1:
```

Comment:
```suggestion
            if (self.columns == self._geometry_column_name).sum() > 1:
```

(similar change as you already did for the check in DataFrame init)